app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,61 @@
+<main>
+<div class="show-recipe container">
+<p style="color: green"><%= notice %></p>
+
+<h1> <%= @recipe.name %></h1>
+
+<div class="recipe-body">
+<%= render @recipe %>
+
+
+
+<%= form_for @recipe, method: :patch do |f| %>
+<div class="swith-btn">
+  <p>Public : </p>
+  <label class="switch">
+  <div><%= f.submit "" %></div>
+    <%= f.check_box :public, { checked: @recipe.public } %>
+    <span class="slider round"></span>
+  </label>
+</div>
+
+<% end %>
+</div>
+<footer class="recipe-footer">
+<div class="footer-buttons">
+<a href= <%= "/shoppings" %>>
+<button>Generate shopping list</button>
+</a>
+
+<a href= <%= "/recipes/#{@recipe.id}/recipe_foods/new" %>>
+<button>Add ingredient</button>
+</a>
+</div>
+<table id="foods">
+  <thead>
+    <tr>
+      <th>Food</th>
+      <th>Quantity</th>
+      <th>Value</th>
+      <th>Actions</th>
+    </tr>
+  </thead>
+    <tbody>
+    <% @recipe.recipe_food.each do |recipe_food| %>
+      <tr>
+          <td><%= recipe_food.food.name %></td>
+          <td><%= recipe_food.quantity %> <%= recipe_food.food.measurement_unit %></td>
+          <td>$ <%= recipe_food.value %></td>
+          <td class="recipe-food-buttons">
+           <%= link_to "Modify", edit_recipe_recipe_food_path(recipe_id: @recipe.id, id: recipe_food.id) %>
+    <%= button_to "Remove", recipe_recipe_food_path(recipe_id: @recipe.id, id: recipe_food.id), method: :delete %>
+          </td>
+      </tr>
+    <% end %>
+  </tbody>
```

Comment:
- [ ] Excellent work so far you have done really well :clap: , however according to the requirements if a recipe does not belong to the current user, he should not be able to take any action such as modify or delete or add ingredients, this is where we have to use authorisation and make sure to give these options only if the user is the owner of the recipe, currently I can modify the public recipe even if i am not the creator of the recipe, I think devise provides a method current_user which you can use to implement authorisation for this please try to fix it :+1: